source/slang/slang-emit-hlsl.cpp

```diff
@@ -673,9 +673,16 @@ void HLSLSourceEmitter::emitVectorTypeNameImpl(IRType* elementType, IRIntegerVal
 
 void HLSLSourceEmitter::emitLoopControlDecorationImpl(IRLoopControlDecoration* decl)
 {
-    if (decl->getMode() == kIRLoopControl_Unroll)
+    switch (decl->getMode())
     {
+    case kIRLoopControl_Unroll:
         m_writer->emit("[unroll]\n");
+        break;
+    case kIRLoopControl_Loop:
+        m_writer->emit("[loop]\n");
+        break;
+    default:
+        SLANG_UNREACHABLE("emit loop control decoration");
```

Comment:
I might leave out the `default` here in case we ever add more loop control modes, where only a subset of them are implemented on each target.